libsolc/libsolc.h

```diff
@@ -57,20 +60,34 @@ char const* solidity_license() SOLC_NOEXCEPT;
 /// The pointer returned must not be freed by the caller.
 char const* solidity_version() SOLC_NOEXCEPT;
 
+/// Allocates a given chunk of memory with @p size bytes.
+///
+/// Use this function inside callbacks to allocate data that is to be passed to the compiler stack,
+/// and use solidity_deallocate() to free this memory again, or solidity_reset() to free it all.
+char* solidity_alloc(size_t _size) SOLC_NOEXCEPT;
+
+/// Explicitly frees the memory that was being allocated with solidity_alloc(size_t size);
+///
+/// Important, this call will abort() in case of any invalid argument being passed to this call.
```

Comment:
What happens if `abort()` is called in emscripten?

Reply:
Will the js environment be invalid after that?

Reply:
@axic? any idea?

Reply:
https://emscripten.org/docs/api_reference/module.html?highlight=abort#Module.onAbort

So perhaps it needs to be reloaded? It is unclear. But `abort` already happens in in case of exceptions in some of the functions.

Reply:
Not as part of this PR - but I wonder whether we could improve the fatal error cases that e.g. in remix just result in "Uncaught js exceptions" by catching all exceptions and calling the read callback one last time with ``kind == "fatal"`` and the best explanation we can provide in ``data`` before aborting...